Content.Server/Blob/BlobCarrierSystem.cs

```diff
@@ -0,0 +1,132 @@
+﻿using Content.Server.Actions;
+using Content.Server.Body.Systems;
+using Content.Server.Ghost.Roles.Components;
+using Content.Server.Mind;
+using Content.Server.Mind.Components;
+using Content.Shared.Actions.ActionTypes;
+using Content.Shared.Blob;
+using Content.Shared.Mobs;
+using Content.Shared.Popups;
+using Robust.Shared.Map;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Timing;
+
+namespace Content.Server.Blob
+{
+    public sealed class BlobCarrierSystem : EntitySystem
+    {
+        [Dependency] private readonly IMapManager _mapManager = default!;
+        [Dependency] private readonly BlobCoreSystem _blobCoreSystem = default!;
+        [Dependency] private readonly MindSystem _mind = default!;
+        [Dependency] private readonly BodySystem _bodySystem = default!;
+        [Dependency] private readonly IPrototypeManager _proto = default!;
+        [Dependency] private readonly ActionsSystem _action = default!;
+        [Dependency] private readonly SharedPopupSystem _popup = default!;
+        [Dependency] private readonly IGameTiming _gameTiming = default!;
+
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<BlobCarrierComponent, MobStateChangedEvent>(OnMobStateChanged);
+            SubscribeLocalEvent<BlobCarrierComponent, TransformToBlobActionEvent>(OnTransformToBlobChanged);
+
+            SubscribeLocalEvent<BlobCarrierComponent, ComponentStartup>(OnStartup);
+            SubscribeLocalEvent<BlobCarrierComponent, ComponentShutdown>(OnShutdown);
+
+            SubscribeLocalEvent<BlobCarrierComponent, MindAddedMessage>(OnMindAdded);
+            SubscribeLocalEvent<BlobCarrierComponent, MindRemovedMessage>(OnMindRemove);
+        }
+
+        private void OnMindAdded(EntityUid uid, BlobCarrierComponent component, MindAddedMessage args)
+        {
+            component.HasMind = true;
+        }
+
+        private void OnMindRemove(EntityUid uid, BlobCarrierComponent component, MindRemovedMessage args)
+        {
+            component.HasMind = false;
+        }
+
+        private void OnTransformToBlobChanged(EntityUid uid, BlobCarrierComponent component, TransformToBlobActionEvent args)
+        {
+            TransformToBlob(uid, component);
+        }
+
+        private void OnStartup(EntityUid uid, BlobCarrierComponent component, ComponentStartup args)
+        {
+            var transformToBlob = new InstantAction(
+                _proto.Index<InstantActionPrototype>("TransformToBlob"));
```

Comment:
Why is this not just using polymorph, an existing, tested system.

Reply:
> Why is this not just using polymorph, an existing, tested system.

was unaware of the presence of the polymorph